modin/engines/base/frame/data.py

```diff
@@ -2280,11 +2284,11 @@ def transpose(self):
             dtypes=new_dtypes,
         )
 
-    def _finalize(self):
+    def finalize(self):
         """
         Perform all deferred calls on partitions.
 
-        This makes `self` Modin DataFrame independent of a history of queries
+        This makes `self` Modin Dataframe independent of a history of queries
         that were used to build it.
         """
-        [part.drain_call_queue() for row in self._partitions for part in row]
+        self._partition_mgr_cls.finalize(self._partitions)
```

Comment:
👍 
this piece was bugging me (even though it was _me_ who implemented it this way), now it looks clean and makes total sense 😄

Reply:
We should always keep the stuff clean :)